fcrepo-search-impl/src/main/java/org/fcrepo/search/impl/DbSearchIndexImpl.java

```diff
@@ -231,6 +317,46 @@ public void addUpdateIndex(final ResourceHeaders resourceHeaders) {
         });
     }
 
+    private void insertRdfTypeAssociations(final List<Long> rdfTypeIds, final Long resourceId) {
+        //add rdf type associations
+        final var jdbcInsertRdfTypeAssociations = new SimpleJdbcInsert(this.jdbcTemplate.getJdbcTemplate());
+        jdbcInsertRdfTypeAssociations.withTableName(SEARCH_RESOURCE_RDF_TYPE_TABLE).usingGeneratedKeyColumns(
+                "id");
+        for (var rdfTypeId : rdfTypeIds) {
+            final var assocParams = new MapSqlParameterSource();
+            assocParams.addValue(RESOURCE_ID_PARAM, resourceId);
+            assocParams.addValue(RDF_TYPE_ID_PARAM, rdfTypeId);
+            jdbcInsertRdfTypeAssociations.execute(assocParams);
+        }
+    }
+
+    private void deleteRdfTypeAssociations(final Long resourceId) {
+        final var deleteParams = new MapSqlParameterSource();
+        deleteParams.addValue(RESOURCE_ID_PARAM, resourceId);
+        jdbcTemplate.update(DELETE_RDF_TYPE_ASSOCIATIONS,
+                deleteParams);
+    }
+
+    private ArrayList<Long> findOrCreateRdfTypesInDb(final List<URI> rdfTypes) {
+        final var jdbcInsertRdfTypes = new SimpleJdbcInsert(this.jdbcTemplate.getJdbcTemplate());
+        jdbcInsertRdfTypes.withTableName(SEARCH_RDF_TYPE_TABLE).usingGeneratedKeyColumns(
+                "id");
```

Comment:
The String `id` is used in a number of places and could be a constant for consistency